packages/flutter/lib/src/cupertino/bottom_tab_bar.dart

```diff
@@ -53,7 +54,11 @@ class CupertinoTabBar extends StatelessWidget implements PreferredSizeWidget {
     this.backgroundColor,
     this.activeColor,
     this.inactiveColor = CupertinoColors.inactiveGray,
-    this.iconSize = 30.0,
+    this.iconSize = 34.0,
```

Comment:
I definitely agree that having useless parameters is something that should be avoided.

However adding a second constructor for this purpose would make relatively simple code way longer than it could be. For example this relatively simple example:

```dart
CupertinoApp(
  title: 'Cupertino demo',
  theme: CupertinoThemeData(),
  home: OrientationBuilder(
    builder: (context, orientation) => CupertinoTabScaffold(
      tabBar: CupertinoTabBar(
        backgroundColor: CupertinoColors.white,
        isWide: orientation == Orientation.landscape || isTablet ? true : false,
        isCompact: orientation == Orientation.landscape && !isTablet ? true : false,
        items: [
          BottomNavigationBarItem(
            icon: Icon(CupertinoIcons.time_solid),
            title: Text('Tab 1')
          ),
          BottomNavigationBarItem(
            icon: Icon(CupertinoIcons.bookmark_solid),
            title: Text('Tab 2')
          ),
        ],
      ),

      ...

    ),
  ),
)
```

...would instead become this:

```dart
CupertinoApp(
  title: 'Cupertino demo',
  theme: CupertinoThemeData(),
  home: OrientationBuilder(
    builder: (context, orientation) => {
      bool isWide = orientation == Orientation.landscape || isTablet ? true : false;
      bool isCompact = orientation == Orientation.landscape && !isTablet ? true : false;

      List<BottomNavigationBarItem> items = [
        BottomNavigationBarItem(
          icon: Icon(CupertinoIcons.time_solid),
          title: Text('Tab 1')
        ),
        BottomNavigationBarItem(
          icon: Icon(CupertinoIcons.bookmark_solid),
          title: Text('Tab 2')
        ),
      ];

      if(isCompact) {
        return CupertinoTabScaffold.compact(
          tabBar: CupertinoTabBar(
            backgroundColor: CupertinoColors.white,
            items: items
          )
        );
      } else if(isWide) {
        return CupertinoTabScaffold.wide(
          tabBar: CupertinoTabBar(
            backgroundColor: CupertinoColors.white,
            items: items
          )
        );
      } else {
        return CupertinoTabScaffold(
          tabBar: CupertinoTabBar(
            backgroundColor: CupertinoColors.white,
            items: items
          )
        );
      }

      ...

    }
  ),
)
```

I guess we can both agree that then the code just becomes unnecessary convoluted.

I'd very much rather replicate what Apple has done and UIKit, and have the tab bar automatically decide whether it should be wide, compact, or normal. As we have seen, by observing iOS's native apps, we can note when it is more correct for the tab bar to assume a specific form. I can provide more detailed notes on my observations in this area if necessary.

As a result, we'd still have 3 different parameters for icon sizes, yes, but it'd more justfied in this case, since the fidelity of the CupertinoTabBar would be improved due to the developers not having to guess that they should/can use a OrientationBuilder with a bunch of screen size calculations to make the tab bar look & feel like the native one does.

Reply:
Yeah you're right, using different constructors to differentiate size classes is awkward. I thought we could do something like this:
```Widget Function(...) tabBar = isCompact ? CupertinoTabBar.compact : CupertinoTabBar.regular;```.
but it turns out constructor tear-offs are not supported yet. 

But I'm still not sure the additional size parameters are required. Wouldn't it also work if we give iconSize different default values based on `isWide` and `isCompact`? I.e.:

```dart
  const CupertinoTabBar({
    Key key,
    @required this.items,
    this.onTap,
    this.currentIndex = 0,
    this.backgroundColor,
    this.activeColor,
    this.inactiveColor = CupertinoColors.inactiveGray,
    double iconSize,
    this.border = const Border(
      top: BorderSide(
        color: _kDefaultTabBarBorderColor,
        width: 0.0, // One physical pixel.
        style: BorderStyle.solid,
      ),
    ),
  }) : assert(items != null),
       assert(
         items.length >= 2,
         "Tabs need at least 2 items to conform to Apple's HIG",
       ),
       assert(currentIndex != null),
       assert(0 <= currentIndex && currentIndex < items.length),
       assert(isCompact != null),
       assert(isWide != null),
       assert(!isCompact || isWide, 'isCompact cannot be true if isWide is not true)),
       this.iconSize = iconSize ?? isWide ? (isCompact ? 22 : 30) : 34,
       assert(inactiveColor != null),
       super(key: key);
```

Reply:
By doing so we're taking away the opportunity to change the icon sizes consistently in case someone wants to retain the automatic layout detection detection